pkg/systemd/quadlet/quadlet.go

```diff
@@ -571,15 +577,18 @@ func ConvertContainer(container *parser.UnitFile, isUser bool, unitsInfoMap map[
 	killMode, ok := service.Lookup(ServiceGroup, "KillMode")
 	if !ok || !(killMode == "mixed" || killMode == "control-group") {
 		if ok {
-			return nil, fmt.Errorf("invalid KillMode '%s'", killMode)
+			return nil, warnings, fmt.Errorf("invalid KillMode '%s'", killMode)
 		}
 
 		// We default to mixed instead of control-group, because it lets conmon do its thing
 		service.Set(ServiceGroup, "KillMode", "mixed")
 	}
 
 	// Read env early so we can override it below
-	podmanEnv := container.LookupAllKeyVal(ContainerGroup, KeyEnvironment)
+	podmanEnv, warn := container.LookupAllKeyVal(ContainerGroup, KeyEnvironment)
+	if warn != nil {
+		warnings = errors.Join(warnings, warn)
+	}
```

Comment:
FYI this != nil  check is not needed as that is already done inside `errors.Join()`
So you can just always call `warnings = errors.Join(warnings, warn)` which should do the exact same thing. Same in the other palaces below. It should make the code a bit simpler.

Reply:
I didn't know that, thanks. Fixed